api/src/main/java/org/openmrs/module/queue/api/QueueRoomService.java

```diff
@@ -14,23 +14,69 @@
 import java.util.List;
 import java.util.Optional;
 
+import org.openmrs.annotation.Authorized;
 import org.openmrs.api.APIException;
 import org.openmrs.module.queue.api.search.QueueRoomSearchCriteria;
 import org.openmrs.module.queue.model.QueueRoom;
+import org.openmrs.module.queue.utils.PrivilegeConstants;
 
 public interface QueueRoomService {
 	
+	/**
+	 * Gets a queue room given uuid.
+	 *
+	 * @param uuid uuid of the queue room to be returned.
+	 * @return {@link org.openmrs.module.queue.model.QueueRoom}
+	 */
+	@Authorized({ PrivilegeConstants.GET_QUEUE_ROOMS })
 	Optional<QueueRoom> getQueueRoomByUuid(@NotNull String uuid);
 	
+	/**
+	 * Gets a queue room by id.
+	 *
+	 * @param id queueRoomId - the id of the queue room to retrieve.
+	 * @return {@link org.openmrs.module.queue.model.QueueRoom}
+	 */
+	@Authorized({ PrivilegeConstants.GET_QUEUE_ROOMS })
 	Optional<QueueRoom> getQueueRoomById(@NotNull int id);
 	
+	/**
+	 * @return {@link List} of all queue rooms
```

Comment:
Though this ticket was not about documenting these methods, you decided to do so and hence should also include the method description. 😊